frontend/src/components/Modal/RiderModalInfo.tsx

```diff
@@ -1,112 +1,183 @@
+                const isValidNeed = (
+                  acc: boolean, 
+                  val: Accessibility
+                ) => acc && Object.values(Accessibility).includes(val);
+                return needsArr.reduce(isValidNeed, true);
+              },
+            })}
           />
-          {errors.email && 'enter some needs'}
+          {errors.needs?.type === "validate" && (
+            <p className={styles.error}>
+              Invalid needs. You can enter 'Assistant', 'Crutches', or
+              'Wheelchair'
+            </p>
+          )}
         </div>
-        <div className={cn(styles.gridR3, styles.gridC2)}>
+        <div className={cn(styles.gridR2, styles.gridCBig2)}>
+          <SRLabel htmlFor='address'>Address: </SRLabel>
           <Input
+            id="address"
             name="address"
             type="text"
             placeholder="Address"
-            ref={register({ required: true })}
+            ref={register({
+              required: true,
+              pattern: /^[a-zA-Z0-9\s,.'-]{3,}$/,
+            })}
           />
-          {errors.email && 'enter an address'}
+          {errors.address && (
+            <p className={styles.error}>Please enter an address</p>
+          )}
         </div>
-        <div className={cn(styles.gridR4, styles.gridC1)}>
-          <Label htmlFor="start">Start Date:</Label>
-          <Input
-            type="date"
-            name="start"
-            ref={register({ required: true })}
-          />
-        </div>
-        <div className={cn(styles.gridR4, styles.gridC2)}>
-          <Label htmlFor="end">End Date:</Label>
-          <Input
-            type="date"
-            name="end"
-            ref={register({ required: true })}
-          />
+        <div className={cn(styles.gridR3, styles.gridCAll)}>
+          <p>Duration</p>
+          <div className={styles.lastRow}>
+            <div>
+              <SRLabel htmlFor='joinDate'>Join Date: </SRLabel>
+              <Input
+                id='joinDate'
+                type="date"
+                name="joinDate"
+                ref={register({ required: true })}
+                className={styles.riderDate}
+              />
+              {errors.joinDate && (
+                <p className={styles.error}>Please enter a join date</p>
+              )}
+            </div>
+            <p className={styles.to}>to</p>
+            <div>
+              <SRLabel htmlFor='endDate'>End Date: </SRLabel>
+              <Input
+                id='endDate'
+                type="date"
+                name="endDate"
+                ref={register({
+                  required: true,
+                  validate: (endDate) => {
+                    const joinDate = getValues("joinDate");
+                    return joinDate < endDate;
+                  },
+                })}
+                className={styles.riderDate}
+              />
+              {errors.end?.type === "required" && (
```

Comment:
You'll either have to change this to errors.endDate?.type or change the name of this to 'end'. Right now the errors don't appear because there is no error for an input of end.